op-service/sources/eth_client.go

```diff
@@ -354,24 +314,21 @@ func (s *EthClient) PayloadByLabel(ctx context.Context, label eth.BlockLabel) (*
 func (s *EthClient) FetchReceipts(ctx context.Context, blockHash common.Hash) (eth.BlockInfo, types.Receipts, error) {
 	info, txs, err := s.InfoAndTxsByHash(ctx, blockHash)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, fmt.Errorf("querying block: %w", err)
 	}
-	// Try to reuse the receipts fetcher because is caches the results of intermediate calls. This means
-	// that if just one of many calls fail, we only retry the failed call rather than all of the calls.
-	// The underlying fetcher uses the receipts hash to verify receipt integrity.
-	var job *receiptsFetchingJob
-	if v, ok := s.receiptsCache.Get(blockHash); ok {
-		job = v
-	} else {
-		txHashes := eth.TransactionsToHashes(txs)
-		job = NewReceiptsFetchingJob(s, s.client, s.maxBatchSize, eth.ToBlockID(info), info.ReceiptHash(), txHashes, s.rethDbPath)
-		s.receiptsCache.Add(blockHash, job)
-	}
-	receipts, err := job.Fetch(ctx)
+
+	txHashes, block := eth.TransactionsToHashes(txs), eth.ToBlockID(info)
+	receipts, err := s.recProvider.FetchReceipts(ctx, block, txHashes)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	if !s.trustRPC {
+		if err := validateReceipts(block, info.ReceiptHash(), txHashes, receipts); err != nil {
+			return info, nil, fmt.Errorf("invalid receipts: %w", err)
+		}
+	}
```

Comment:
I don't think we currently have a unit test that verified the `validateReceipts` method actually gets called. It could be a follow up PR but with this refactor it would hopefully be much easier to test that so would be good to add the missing test if we can.

Reply:
Added test `TestEthClient_validateReceipts`